Tests/BackendIntegrationTests/BaseStoreKitIntegrationTests.swift

```diff
@@ -37,7 +37,8 @@ class BaseStoreKitIntegrationTests: BaseBackendIntegrationTests {
             try self.configureTestSession()
         }
 
-        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *),
+           self.shouldCleanUpTransactionsBeforeTest {
```

Comment:
🤦🏻  this meant that we were always clearing up transactions in observer mode tests leading to potentially false positives.
Luckily they still pass with this!